src/app/organisms/emoji-board/recent.js

```diff
@@ -8,10 +8,15 @@ function getRecentEmojisRaw() {
 }
 
 export function getRecentEmojis(limit) {
-  return getRecentEmojisRaw()
+  const res = [];
+  getRecentEmojisRaw()
     .sort((a, b) => b[1] - a[1])
-    .slice(0, limit)
-    .map(([unicode]) => emojis.find((e) => e.unicode === unicode));
+    .find(([unicode]) => {
+      const emoji = emojis.find((e) => e.unicode === unicode);
+      if (emoji) return res.push(emoji) >= limit;
+      return false;
+    });
```

Comment:
I am not sure what's happening here. we are using `find` but it's doing `forEach` work. also line 16 is confusing to me can you please explain it?

Reply:
The `find` works as a `forEach` but breaks when `true` is being returned. That happens when after an emoji gets inserted into the list it hits the limit (of 3 * 7 emojis).

`res.push()` returns the new array length. It would be equivalent to writing:
```js
if (emoji) {
  res.push(emoji);
  return res.length >= limit;
}
```